tensorrt_llm/_torch/auto_deploy/transform/library/attention.py

```diff
@@ -414,91 +325,242 @@ def _grouped_attn_replacement_4(q, k, v, attn_mask, dropout_p, scale):
+    Match and replace the repeat_kv pattern with torch.ops.auto_deploy.torch_attention_repeat_kv.
     """
-    Match and transform attention operations to match the layout expected by the attention backend.
 
-    If the attention backend expects 'bnsd' layout (batch, num_heads, seq_len, head_dim), which
-    is the default for SDPA operations, we don't need to transform anything.
+    def _apply(
+        self, gm: GraphModule, cm: CachedSequenceInterface, factory: ModelFactory
+    ) -> Tuple[GraphModule, TransformInfo]:
+        def register_repeat_kv(patterns: ADPatternMatcherPass):
+            dummy_args = [
+                torch.randn(8, 8, 16, 64, device="cuda", dtype=torch.float16),
+                7,
+            ]
+            register_ad_pattern(
+                search_fn=_repeat_kv_pattern,
+                replace_fn=_repeat_kv_repl,
+                patterns=patterns,
+                dummy_args=dummy_args,
+                op_ignore_types={
+                    torch.ops.aten.reshape.default: (int,),
+                    torch.ops.aten.expand.default: (int,),
+                },
+                scalar_workaround={"n_rep": dummy_args[1]},
+            )
 
-    If the backend expects 'bsnd' layout (batch, seq_len, num_heads, head_dim), we insert
-    appropriate transposes before and after SDPA operations and replace them with bsnd_grouped_sdpa.
+        num_kv_patterns = _apply_pattern(gm, "Repeat KV", register_repeat_kv, shape_prop=True)
+
+        info = TransformInfo(
+            skipped=False,
+            num_matches=num_kv_patterns,
+            is_clean=True,
+            has_valid_shapes=True,
+        )
+
+        return gm, info
+
+
+@TransformRegistry.register("match_eager_attention")
+class MatchEagerAttention(BaseTransform):
+    """
+    Match and replace the eager attention pattern with torch.ops.auto_deploy.torch_attention_sdpa.
     """
-    # Get attention layout from attention_op
-    attention_layout = attention_op.get_attention_layout()
 
-    # List of SDPA operations to look for
-    sdpa_ops = {
-        torch.ops.auto_deploy.torch_attention_sdpa,
-        torch.ops.auto_deploy.torch_attention_grouped_sdpa,
-    }
+    def _apply(
+        self, gm: GraphModule, cm: CachedSequenceInterface, factory: ModelFactory
+    ) -> Tuple[GraphModule, TransformInfo]:
+        def register_eager_attention(patterns: ADPatternMatcherPass):
+            for pattern_config in _get_sfdp_patterns():
+                register_ad_pattern(**pattern_config, patterns=patterns)
 
-    graph = gm.graph
-    num_bsnd_patterns = 0
+        num_eager_patterns = _apply_pattern(gm, "Eager Attention", register_eager_attention)
 
-    # Look for SDPA operations
-    for sdpa_node in list(graph.nodes):
-        if sdpa_node.op != "call_function" or not is_op(sdpa_node, sdpa_ops):
-            continue
+        info = TransformInfo(
+            skipped=False,
+            num_matches=num_eager_patterns,
+            is_clean=True,
```

Comment:
```suggestion
            is_clean=False,
```

I don't think we can guarantee a clean graph after this. Better to set it to false and then have  `canonicalize_graph` be triggered in the wrapper